providers/base/bin/suspend_stats.py

```
@@ -0,0 +1,143 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+import argparse
+import sys
+import os
+
+
+class SuspendStats:
+    """
+    This class is used to parse the information under
+    /sys/power/suspend_stats/
+
+    """
+
+    contents = {}
+
+    def __init__(self):
+        suspend_stat_path = "/sys/power/suspend_stats/"
+        self.contents = self.collect_content_under_directory(suspend_stat_path)
+
+    def collect_content_under_directory(self, search_directory: str) -> dict:
+        """
+        Collect all content under specific directory by filename
+
+        :param search_directory: The directory to search through.
+
+        :returns: collected content by filename
+        """
+        content = {}
+        for root, dirs, files in os.walk(search_directory):
+            for file_name in files:
+                file_path = os.path.join(root, file_name)
+                with open(
+                    file_path, "r", encoding="utf-8", errors="ignore"
+                ) as file:
+                    content[file_name] = file.read().splitlines()[0]
+        return content
+
+    def print_all_content(self):
+        """
+        Print all contents under suspend_stats
+
+        """
+        for c in self.contents:
+            print("{}:{}".format(c, self.contents[c]))
+
+    def is_after_suspend(self) -> bool:
+        """
+        The system is under after suspend status or not
+
+        :returns: return Ture while system is under after suspend status
+        """
+        return (
+            self.contents["success"] != "0"
+            and self.contents["failed_prepare"] == "0"
+            and self.contents["failed_suspend"] == "0"
+            and self.contents["failed_resume"] == "0"
+        )
```

Big questoion is, why are we comparing these values as strings? Is it even possible that these are not valid integers? if that happens I suppose its a big deal, for this reason I've also suggested to remove the utf8/ignore errors above, if sysfs is broken we want to know

Another big question is what do we do if one or more of these files is missing? is it possible? if it is, when does it happen? should we have a default for those situations?